Sources/MapboxNavigation/CarPlaySearchController.swift

```diff
@@ -1,13 +1,123 @@
 #if canImport(CarPlay)
+import Foundation
 import CarPlay
+import MapboxGeocoder
 import MapboxDirections
 
 @available(iOS 12.0, *)
-public protocol CarPlaySearchControllerDelegate: class {
+public protocol CarPlaySearchControllerDelegate: class, CPSearchTemplateDelegate {
     func previewRoutes(to waypoint: Waypoint, completionHandler: @escaping () -> Void)
     func resetPanButtons(_ mapTemplate: CPMapTemplate)
     func pushTemplate(_ template: CPTemplate, animated: Bool)
     func popTemplate(animated: Bool)
+
+    var recentItems: [RecentItem] { get set }
+    var recentSearchItems: [CPListItem]? { get set }
+    var recentSearchText: String? { get set }
+    
+    func searchTemplate(_ searchTemplate: CPSearchTemplate, updatedSearchText searchText: String, completionHandler: @escaping ([CPListItem]) -> Void)
+    func searchTemplate(_ searchTemplate: CPSearchTemplate, selectedResult item: CPListItem, completionHandler: @escaping () -> Void)
+    func forwardGeocodeOptions(_ searchText: String) -> ForwardGeocodeOptions
+    func selectResult(item: CPListItem, completionHandler: @escaping () -> Void)
+    func recentSearches(_ searchText: String) -> [CPListItem]
+    func resultsOrNoResults(_ items: [CPListItem], limit: UInt?) -> [CPListItem]
```

Comment:
These methods overuse the `_` keyword. A method name should only omit the first argument’s label if the first part of the method name (e.g., “forwardGeocodeOptions”) describes the first argument. Otherwise, use a preposition or the local variable name as the argument label.